# Data preparation

```{r packs, warning=FALSE, comments="", message=FALSE}
library(tidyverse)
library(ape)
```


## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("data/DMB0173_metadata.tsv.gz") %>%
  dplyr::rename(sample = EHI_plaintext,
                host_order = order,
                host_species = species)
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv("data/DMB0173_counts.tsv.gz") %>%
  rename(genome=1) %>%
  select(c("genome", sample_metadata$sample))
genomes <- read_counts$genome
```

### Genome coverage

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_coverage <- read_tsv("data/DMB0173_coverage.tsv.gz") %>%
  rename(genome=1) %>%
  select(c("genome", sample_metadata$sample))%>%
  arrange(match(genome, genomes))
```

### Genome metadata

```{r load_genome_metadata, warning=FALSE, comments="", message=FALSE}
mag_extra <- read_delim("data/mag_extra_metadata.csv.gz") %>%
  mutate(mag_name = str_replace(mag_name, ".fa", ""))

genome_metadata <- read_tsv("data/DMB0173_mag_info.tsv.gz")%>%
  rename(length = mag_size) %>%
  mutate(phylum = str_remove_all(phylum, "p__")) %>%
  inner_join(., mag_extra, by = join_by(genome == mag_name)) %>%
  arrange(match(genome, genomes))
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("data/DMB0173.tree.gz")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label, "'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip = genome_metadata$genome) # keep only MAG tips
```


## Create working objects

Transform the original data files into working objects for downstream analyses.

### Filter reads by evenness of coverage (0.5 = 50% of the genome is covered by reads)

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
min_coverage=0.5
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts_filt %>% #u
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r remove_low_smf_samples, warning=FALSE, comments="", message=FALSE}

low_smf_samples <- sample_metadata %>%
  filter(singlem_fraction < 30)

low_smf_samples %>% 
  tt()

#perform filtering
genome_counts_filt <- genome_counts_filt %>%
  select(!low_smf_samples$sample)

read_counts <- read_counts %>%
  select(!low_smf_samples$sample)

```


### Distill annotations into GIFTs 

```{r distill_annotations, warning=FALSE, comments="", message=FALSE}
load("data/genome_gifts_raw.Rdata")
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts_raw),] 
rownames(genome_counts_filt) <- NULL
genome_gifts <- genome_gifts_raw[rownames(genome_gifts_raw) %in% genome_counts_filt$genome,] 
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate(phylum=str_remove_all(phylum, "p__")) %>%
  right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
  arrange(match(genome, genome_tree$tip.label)) %>%
  select(phylum, colors) %>% 
  unique() %>%
  arrange(phylum) %>%
  pull(colors, name=phylum)
```

## Prepare a phyloseq object

```{r phyloseq_objects, warning=FALSE, comments="", message=FALSE}
phylo_samples <- sample_metadata %>% 
  column_to_rownames("sample") %>% 
  sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  column_to_rownames("genome") %>% 
  as.matrix() %>% 
  tax_table() #convert to phyloseq tax_table object

physeq_genome <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_clr <- microbiome::transform(physeq_genome, 'clr')

```

## Prepare colour codes for host species

```{r set_colours, warning=FALSE, comments="", message=FALSE}

species_colours <- c(
  "Podarcis pityusensis" = "#bef7ff", 
  "Podarcis gaigeae" = "#a6e2ff", 
  "Podarcis milensis" = "#8eccff", 
  "Podarcis filfolensis" = "#75b7ff", 
  "Podarcis liolepis" = "#5da1ff", 
  "Podarcis muralis" = "#458cff",
  "Iberolacerta bonnali" = "#c080ff",
  "Iberolacerta aranica" = "#8000ff",
  "Lacerta viridis" = "#804280",
  "Chalcides striatus" = "#80d9ff",
  "Timon lepidus" = "#797980",
  "Psammodromus algirus" = "#a1a1aa",
  "Rattus rattus" = "#ff0000",
  "Rattus lutreolus" = "#ff5555",
  "Myodes glareolus" = "#aa6c00",
  "Myodes rufocanus" = "#ffa200",
  "Apodemus sylvaticus" = "#5eff00",
  "Apodemus flavicollis" = "#4ed500",
  "Peromyscus maniculatus" = "#ffa7a7",
  "Muscardinus avellanarius" = "#ff416d",
  "Sciurus vulgaris" = "#00ff91"
)




```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     physeq_genome,
     physeq_genome_clr,
     species_colours,
     file = "data/data.Rdata")
```