---
title: "AlberdiLab | EHI Squamate Rodent metagenomics"
subtitle: "EHI Squamate Rodent"
author:
  - Raphael Eisenhofer^[University of Copenhagen, raphael.eisenhofer@sund.ku.dk]

  - Antton Alberdi^[University of Copenhagen, antton.alberdi@sund.ku.dk]
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
url: https://alberdilab.github.io/EHI_squamate_rodent
description: |
  Data analysis code for the study on comparing squamates and rodent gut metagenomes
link-citations: yes
github-repo: alberdilab/EHI_squamate_rodent
---

# Introduction


```{r knitr_opts, echo=FALSE}
knitr::opts_chunk$set(
    class.source = "script-source",
    class.output = "script-output",
    comment = NA)
```


This webbook contains all the code used for data analysis in study on comparing squamate and rodent gut metagenomes from the Earth Hologenome Initiative.

## Prepare the R environment

### Environment

To reproduce all the analyses locally, clone this repository in your computer using:

```
RStudio > New Project > Version Control > Git
```

And indicating the following git repository:

> https://github.com/alberdilab/EHI_squamate_rodent.git

Once the R project has been created, follow the instructions and code chunks shown in this webbook.

### Libraries

The following R packages are required for the data analysis.

```{r load_libraries, warning=FALSE, comments="", message=FALSE}
# Base
library(rmarkdown)
library(R.utils)
library(knitr)
library(tidyverse)
library(devtools)
library(tinytable)
library(broom)
library(broom.mixed)
library(janitor)

# For tree handling
library(ape)
library(phyloseq)
library(phytools)

# For plotting
library(plotly)
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(ggnewscale)
library(gridExtra)
library(ggtreeExtra)
library(ggtree)
library(ggh4x)

# For statistics
library(spaa)
library(vegan)
library(Rtsne)
library(geiger)
library(hilldiv2)
library(distillR)
library(ANCOMBC)
library(lme4)
library(matrixStats)
```

<!--chapter:end:index.Rmd-->

# Data preparation

```{r packs, warning=FALSE, comments="", message=FALSE}
library(tidyverse)
library(ape)
```


## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("data/DMB0173_metadata.tsv.gz") %>%
  dplyr::rename(sample = EHI_plaintext,
                host_order = order,
                host_species = species)
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv("data/DMB0173_counts.tsv.gz") %>%
  rename(genome=1) %>%
  select(c("genome", sample_metadata$sample))
genomes <- read_counts$genome
```

### Genome coverage

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_coverage <- read_tsv("data/DMB0173_coverage.tsv.gz") %>%
  rename(genome=1) %>%
  select(c("genome", sample_metadata$sample))%>%
  arrange(match(genome, genomes))
```

### Genome metadata

```{r load_genome_metadata, warning=FALSE, comments="", message=FALSE}
mag_extra <- read_delim("data/mag_extra_metadata.csv.gz") %>%
  mutate(mag_name = str_replace(mag_name, ".fa", ""))

genome_metadata <- read_tsv("data/DMB0173_mag_info.tsv.gz")%>%
  rename(length = mag_size) %>%
  mutate(phylum = str_remove_all(phylum, "p__")) %>%
  inner_join(., mag_extra, by = join_by(genome == mag_name)) %>%
  arrange(match(genome, genomes))
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("data/DMB0173.tree.gz")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label, "'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip = genome_metadata$genome) # keep only MAG tips
```


## Create working objects

Transform the original data files into working objects for downstream analyses.

### Filter reads by evenness of coverage (0.5 = 50% of the genome is covered by reads)

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
min_coverage=0.5
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts_filt %>% #u
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, warning=FALSE, comments="", message=FALSE}
load("data/genome_gifts_raw.Rdata")
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts_raw),] 
rownames(genome_counts_filt) <- NULL
genome_gifts <- genome_gifts_raw[rownames(genome_gifts_raw) %in% genome_counts_filt$genome,] 
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
  mutate(phylum=str_remove_all(phylum, "p__")) %>%
  right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
  arrange(match(genome, genome_tree$tip.label)) %>%
  select(phylum, colors) %>% 
  unique() %>%
  arrange(phylum) %>%
  pull(colors, name=phylum)
```

## Prepare a phyloseq object

```{r phyloseq_objects, warning=FALSE, comments="", message=FALSE}
phylo_samples <- sample_metadata %>% 
  column_to_rownames("sample") %>% 
  sample_data() #convert to phyloseq sample_data object
phylo_genome <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  otu_table(., taxa_are_rows = TRUE)
phylo_taxonomy <- genome_metadata %>%
  column_to_rownames("genome") %>% 
  as.matrix() %>% 
  tax_table() #convert to phyloseq tax_table object

physeq_genome <- phyloseq(phylo_genome, phylo_taxonomy, phylo_samples)
physeq_genome_clr <- microbiome::transform(physeq_genome, 'clr')

```

## Prepare colour codes for host species

```{r set_colours, warning=FALSE, comments="", message=FALSE}

species_colours <- c(
  "Podarcis pityusensis" = "#bef7ff", 
  "Podarcis gaigeae" = "#a6e2ff", 
  "Podarcis milensis" = "#8eccff", 
  "Podarcis filfolensis" = "#75b7ff", 
  "Podarcis liolepis" = "#5da1ff", 
  "Podarcis muralis" = "#458cff",
  "Iberolacerta bonnali" = "#c080ff",
  "Iberolacerta aranica" = "#8000ff",
  "Lacerta viridis" = "#804280",
  "Chalcides striatus" = "#80d9ff",
  "Timon lepidus" = "#797980",
  "Psammodromus algirus" = "#a1a1aa",
  "Rattus rattus" = "#ff0000",
  "Rattus lutreolus" = "#ff5555",
  "Myodes glareolus" = "#aa6c00",
  "Myodes rufocanus" = "#ffa200",
  "Apodemus sylvaticus" = "#5eff00",
  "Apodemus flavicollis" = "#4ed500",
  "Peromyscus maniculatus" = "#ffa7a7",
  "Muscardinus avellanarius" = "#ff416d",
  "Sciurus vulgaris" = "#00ff91"
)




```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     physeq_genome,
     physeq_genome_clr,
     species_colours,
     file = "data/data.Rdata")
```

<!--chapter:end:01_prepare_data.Rmd-->

# Data statistics

```{r load_data_stats}
load("data/data.Rdata")
```

## filter samples with high host data
```{r load_data_host_filt, comment="", message=FALSE, warning=FALSE, eval=FALSE}

read_counts <- read_counts %>% 
  select(one_of(c("genome", sample_metadata$sample))) %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., read_counts, by = "genome") %>% 
  arrange(match(genome,read_counts$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips

#load("data/genome_gifts.Rdata")
```

## Sequencing reads statistics

```{r reads_stats}
sample_metadata %>% 
    summarise(Total=sum(reads_post_fastp * 150 / 1000000000) %>% round(2), 
              mean=mean(reads_post_fastp * 150 / 1000000000) %>% round(2),
              sd=sd(reads_post_fastp * 150 / 1000000000) %>% round(2)) %>%
    unite("Average",mean, sd, sep = " Â± ", remove = TRUE) %>%
    tt()
```

## DNA fractions
```{r dna_fractions_stats}
sequence_fractions <- read_counts %>%
  pivot_longer(-genome, names_to = "sample", values_to = "value") %>%
  group_by(sample) %>%
  summarise(mags = sum(value)) %>%
	left_join(sample_metadata, by = join_by(sample == sample)) %>%
	select(sample, mags, metagenomic_bases, host_bases, bases_lost_fastp_percent) %>%
	mutate(mags_bases = mags*146) %>%
	mutate(lowqual_bases = ((metagenomic_bases+host_bases)/(1-bases_lost_fastp_percent))-(metagenomic_bases+host_bases)) %>%
	mutate(unmapped_bases = metagenomic_bases - mags_bases) %>%
	mutate(unmapped_bases = ifelse(unmapped_bases < 0, 0, unmapped_bases)) %>%
	select(sample, lowqual_bases, host_bases, unmapped_bases, mags_bases)

sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  tt()

sequence_fractions %>%
  mutate_at(vars(-sample), ~./1000000000) %>%
  rename("Sample"=1, "Low quality"=2, "Mapped to host"=3, "Unmapped"=4, "Mapped to MAGs"=5) %>%
  summarise(across(where(is.numeric), mean, na.rm = TRUE)) %>%
  tt()

```


```{r dna_fractions_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
sequence_fractions %>%
	pivot_longer(!sample, names_to = "fraction", values_to = "value") %>%
	mutate(value = value / 1000000000) %>%
	mutate(fraction = factor(fraction, levels = c("lowqual_bases", "host_bases", "unmapped_bases", "mags_bases"))) %>%
  
	ggplot(., aes(x = sample, y = value, fill=fraction)) +
	    geom_bar(position="stack", stat = "identity") +
      scale_fill_manual(name="Sequence type",
                    breaks=c("lowqual_bases","host_bases","unmapped_bases","mags_bases"),
                    labels=c("Low quality","Mapped to host","Unmapped","Mapped to MAGs"),
                    values=c("#CCCCCC", "#bcdee1", "#d8b8a3","#93655c"))+
	    labs(x = "Samples", y = "Amount of data (GB)") +
	    theme_classic() +
	    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "bottom")
```

## Recovered microbial fraction

```{r data_estimations_plot, message=FALSE, warning=FALSE, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
singlem_table <- sequence_fractions %>%
	mutate(mags_proportion = round((mags_bases / (mags_bases + unmapped_bases))*100,2)) %>%
  left_join(sample_metadata, by = join_by(sample == sample))  %>%
	mutate(singlem_proportion = round(singlem_fraction,2)) %>%
	select(sample,mags_proportion,singlem_proportion) %>%
	mutate(mags_proportion = ifelse(singlem_proportion == 0, 0, mags_proportion)) %>% #convert zeros to NA
	mutate(singlem_proportion = ifelse(singlem_proportion == 0, NA, singlem_proportion)) %>% #convert zeros to NA
	mutate(singlem_proportion = ifelse(singlem_proportion < mags_proportion, NA, singlem_proportion)) %>% #if singlem is smaller, then NA, to simplify plot
	mutate(singlem_proportion = ifelse(singlem_proportion > 100, 100, singlem_proportion)) #simplify

singlem_table %>%
	pivot_longer(!sample, names_to = "proportion", values_to = "value") %>%
	mutate(proportion = factor(proportion, levels = c("mags_proportion","singlem_proportion"))) %>%
	ggplot(., aes(x = value, y = sample, color=proportion)) +
			geom_line(aes(group = sample), color = "#f8a538") +
			geom_point() +
      scale_color_manual(name="Proportion",
                    breaks=c("mags_proportion","singlem_proportion"),
                    labels=c("Recovered","Estimated"),
                    values=c("#52e1e8", "#876b53"))+
			theme_classic() +
			labs(y = "Samples", x = "Prokaryotic fraction (%)") +
	    theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size=6),legend.position = "right")

```

```{r damr, message=FALSE, warning=FALSE}
damr <- singlem_table %>%
  mutate(damr=ifelse(is.na(singlem_proportion),100,mags_proportion/singlem_proportion*100)) %>%
  select(sample,damr)

damr %>% 
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()

damr %>% 
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()

damr %>% 
  tt()

damr %>% 
  filter(damr>95) %>%
  nrow()

damr %>% 
  summarise(mean=mean(damr),sd=sd(damr)) %>% 
  tt()
```


<!--chapter:end:02_data_statistics.Rmd-->

# MAG catalogue

```{r load_data_mag, message=FALSE, warning=FALSE}
load("data/data.Rdata")
```

## filter samples with high host data
```{r load_data_mag_filt, comment="", message=FALSE, warning=FALSE, eval=FALSE}

genome_counts_filt <- genome_counts %>% 
    select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))

genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips

#load("data/genome_gifts.Rdata")
```

## Genome phylogeny

```{r genome_phylogeny, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}
# Generate the phylum color heatmap
phylum_heatmap <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(genome,phylum) %>%
    mutate(phylum = factor(phylum, levels = unique(phylum))) %>%
    column_to_rownames(var = "genome")

# Generate new species table
newspecies_table <- genome_metadata %>% 
  mutate(newspecies=ifelse(species=="s__","Y","N")) %>% 
  select(genome,newspecies) %>% 
  column_to_rownames(var = "genome")
  
# Generate table
heatmap <- genome_counts_filt  %>%
  pivot_longer(!genome,names_to="sample",values_to="abundance") %>% 
  left_join(sample_metadata,by="sample") %>% 
  group_by(genome) %>% 
  summarise(presence=ifelse(sum(abundance)>0,"present","absent")) %>% 
  column_to_rownames(var="genome")


# Generate  basal tree
circular_tree <- force.ultrametric(genome_tree, method="extend") %>% # extend to ultrametric for the sake of visualisation
    ggtree(., layout="fan", open.angle=10, size=0.2)

# Add phylum ring
circular_tree <- gheatmap(circular_tree, phylum_heatmap, offset=0.05, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=phylum_colors) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Flush color scale to enable a new color scheme in the next ring
circular_tree <- circular_tree + new_scale_fill()

# Add completeness ring
circular_tree <- circular_tree +
        new_scale_fill() +
        scale_fill_gradient(low = "#d1f4ba", high = "#f4baba") +
        geom_fruit(
                data=genome_metadata,
                geom=geom_bar,
                mapping = aes(x=completeness, y=genome, fill=contamination),
                offset = 0.3,
                pwidth = 0.1,
                orientation="y",
              stat="identity")

# Add genome-size ring
circular_tree <- circular_tree + new_scale_fill()

circular_tree <- gheatmap(circular_tree, heatmap, offset=0.3, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#ffffff","#74C8AE")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0)) +
        new_scale_fill()

circular_tree <- gheatmap(circular_tree, newspecies_table, offset=1.2, width=0.05, colnames=FALSE) +
        scale_fill_manual(values=c("#f4f4f4","#666666")) +
        theme(legend.position = "none", plot.margin = margin(0, 0, 0, 0), panel.margin = margin(0, 0, 0, 0))

# Add text
circular_tree <-  circular_tree +
        annotate('text', x=2.9, y=0, label='Phylum', family='arial', size=3.5) +
        annotate('text', x=3.3, y=0, label='Group', family='arial', size=3.5) +
        annotate('text', x=3.8, y=0, label='Genome quality', family='arial', size=3.5) +
        annotate('text', x=4.2, y=0, label='New species', family='arial', size=3.5)

#Plot circular tree
circular_tree %>% open_tree(30) %>% rotate_tree(90)
```

## Taxonomy overview

```{r genome_taxonomy_bacteria, message=FALSE, warning=FALSE,}
tax_mag <-genome_metadata %>% 
  group_by(phylum) %>%
  summarise(mag_n=n()) 

tax_mag %>%
  mutate(percetage_mag=round(mag_n*100/sum(mag_n), 2)) %>% 
  arrange(-percetage_mag) %>%
  tt()
```

## Mag size (MB)

```{r correct_mag_size, message=FALSE, warning=FALSE}
genome_metadata <- genome_metadata %>% 
  mutate(corrected_size = 100 * length / completeness)
```

Mags average size (MB)
```{r mag_size_all_mean, message=FALSE, warning=FALSE}
genome_metadata %>% 
  summarise(Average_corrected_size=mean(corrected_size)) %>% 
  tt()
```

Minimum Mags size (MB)

```{r mag_size_min,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==min(corrected_size)) %>% 
  tt()
```

Maximum Mags size (MB)

```{r mag_size_max,message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(corrected_size==max(corrected_size)) %>% 
  tt()
```

Mags arrange by size (MB)
```{r mag_size_all, message=FALSE, warning=FALSE}
genome_metadata %>% 
  arrange(corrected_size) %>% 
  paged_table()
```

Mags average size and completeness by phylum

```{r mag_size_phylum, message=FALSE, warning=FALSE}
genome_metadata %>% 
  group_by(phylum) %>%
  summarise(average_size=mean(corrected_size),
            sd_size=sd(corrected_size),
            average_comp=mean(completeness),
            sd_comp=sd(completeness)) %>%
  mutate(Size=str_c(round(average_size,2),"Â±",round(sd_size,2)),
         Completeness=str_c(round(average_comp,2),"Â±",round(sd_comp,2))) %>% 
  arrange(-average_size) %>% 
  select(phylum, Size, Completeness) %>% 
  tt()
```


## Genome quality

```{r genome_quality, message=FALSE, warning=FALSE}
genome_metadata %>% 
    summarise(completeness_mean=mean(completeness) %>% round(2) %>% as.character(), 
              completeness_sd=sd(completeness) %>% round(2) %>% as.character(), 
              contamination_mean=mean(contamination) %>% round(2), 
              contamination_sd=sd(contamination) %>% round(2)) %>%
    unite("Completeness",completeness_mean, completeness_sd, sep = " Â± ", remove = TRUE) %>%
    unite("Contamination",contamination_mean, contamination_sd, sep = " Â± ", remove = TRUE) %>%
    tt()
```

#### Genomes per host species

```{r completeness_per_species, message=FALSE, warning=FALSE}
genome_metadata %>%
  ggplot(aes(x = host_species, y = completeness, colour = host_species)) +
  geom_jitter(size = 2, alpha = 0.3, height = 0, width = 0.2) + 
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(x = "", y = "CheckM2 completeness score")

```
```{r contaimation_per_species, message=FALSE, warning=FALSE}
genome_metadata %>%
  ggplot(aes(x = host_species, y = contamination, colour = host_species)) +
  geom_jitter(size = 2, alpha = 0.3, height = 0, width = 0.2) + 
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(x = "", y = "CheckM2 contamination score")

```

#### Genome statistics per host species 

```{r genome_stats_by_species, message=FALSE, warning=FALSE}

genome_metadata %>%
  group_by(host_species) %>%
  mutate(n_mags = n()) %>%
  ungroup() %>%
  ggplot(aes(x = host_species, y = n_mags, colour = host_species)) +
  geom_jitter(size = 2, alpha = 0.3, height = 0, width = 0.2) + 
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  scale_y_continuous(limits = c(75, 450)) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(x = "", y = "Number of MAGs")

```



```{r genome_quality_plot, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}

#Generate quality biplot
genome_biplot <- genome_metadata %>%
  select(c(genome,domain,phylum,completeness,contamination,length)) %>%
  arrange(match(genome, rev(genome_tree$tip.label))) %>% #sort MAGs according to phylogenetic tree
  ggplot(aes(x=completeness,y=contamination,size=length,color=phylum)) +
              geom_point(alpha=0.7) +
                    ylim(c(10,0)) +
                    scale_color_manual(values=phylum_colors) +
                    labs(y= "Contamination", x = "Completeness") +
                    theme_classic() +
                    theme(legend.position = "none")

#Generate contamination boxplot
genome_contamination <- genome_metadata %>%
            ggplot(aes(y=contamination)) +
                    ylim(c(10,0)) +
                    geom_boxplot(colour = "#999999", fill="#cccccc") +
                    theme_void() +
                    theme(legend.position = "none",
                        axis.title.x = element_blank(),
                        axis.title.y = element_blank(),
                        axis.text.y=element_blank(),
                        axis.ticks.y=element_blank(),
                        axis.text.x=element_blank(),
                        axis.ticks.x=element_blank(),
                        plot.margin = unit(c(0, 0, 0.40, 0),"inches")) #add bottom-margin (top, right, bottom, left)

#Generate completeness boxplot
genome_completeness <- genome_metadata %>%
        ggplot(aes(x=completeness)) +
                xlim(c(50,100)) +
                geom_boxplot(colour = "#999999", fill="#cccccc") +
                theme_void() +
                theme(legend.position = "none",
                    axis.title.x = element_blank(),
                    axis.title.y = element_blank(),
                    axis.text.y=element_blank(),
                    axis.ticks.y=element_blank(),
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    plot.margin = unit(c(0, 0, 0, 0.50),"inches")) #add left-margin (top, right, bottom, left)

#Render composite figure
grid.arrange(grobs = list(genome_completeness,genome_biplot,genome_contamination),
        layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,1,4),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3),
                              c(2,2,2,2,2,2,2,2,2,2,2,3)))

```


## Functional overview


```{r function_heatmap, message=FALSE, warning=FALSE, fig.height=10, fig.width=10, fig.fullwidth=TRUE}

# Aggregate basal GIFT into functions
GIFTs_elements <- to.elements(genome_gifts, GIFT_db)

function_table <- GIFTs_elements %>%
    to.functions(., GIFT_db)

# Generate  basal tree
function_tree <- force.ultrametric(genome_tree, method="extend") %>%
                ggtree(., size = 0.3) 

#Add phylum colors next to the tree tips
function_tree <- gheatmap(function_tree, phylum_heatmap, offset=0, width=0.1, colnames=FALSE) +
            scale_fill_manual(values=phylum_colors) +
            labs(fill="Phylum")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

#Add functions heatmap
function_tree <- gheatmap(function_tree, function_table, offset=0.5, width=3.5, colnames=FALSE, color = NA) +
            vexpand(.08) +
            coord_cartesian(clip = "off") +
            scale_fill_gradient(low = "#f4f4f4", high = "steelblue", na.value="white") +
            labs(fill="GIFT")

#Reset fill scale to use a different colour profile in the heatmap
function_tree <- function_tree + new_scale_fill()

# Add completeness barplots
function_tree <- function_tree +
            geom_fruit(data=genome_metadata,
            geom=geom_bar,
            grid.params=list(axis="x", text.size=2, nbreak = 1),
            axis.params=list(vline=TRUE),
            mapping = aes(x=length, y=genome, fill=completeness),
                 offset = 3.8,
                 orientation="y",
                 stat="identity") +
            scale_fill_gradient(low = "#cf8888", high = "#a2cc87") +
            labs(fill="Genome\ncompleteness")

function_tree
```

## Functional ordination (at DistillR function-level)

```{r function_ordination, message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
# Generate the tSNE ordination
tSNE_function <- Rtsne(X=function_table, dims = 2, check_duplicates = FALSE)

# Plot the ordination
function_ordination <- tSNE_function$Y %>%
                as.data.frame() %>%
                mutate(genome=rownames(function_table)) %>%
                inner_join(genome_metadata, by="genome") %>%
                rename(tSNE1="V1", tSNE2="V2") %>%
                select(genome,phylum,tSNE1,tSNE2, length) %>%
                ggplot(aes(x = tSNE1, y = tSNE2, color = phylum, size=length))+
                            geom_point(shape=16, alpha=0.7) +
                            scale_color_manual(values=phylum_colors) +
                            theme_minimal() +
                labs(color="Phylum", size="Genome size") +
                guides(color = guide_legend(override.aes = list(size = 5))) # enlarge Phylum dots in legend

function_ordination
```

<!--chapter:end:03_mag_catalogue.Rmd-->

# Community composition

## Filter data

```{r load_data_community, comment="", message=FALSE, warning=FALSE, eval=FALSE}
load("data/data.Rdata")
```

Filter samples with high host data
```{r load_data_host_filtering, comment="", message=FALSE, warning=FALSE, eval=FALSE}

genome_counts_filt <- genome_counts %>%
  select(one_of(c("genome",sample_metadata$sample)))%>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))
genome_counts <- genome_counts_filt
genome_metadata <- genome_metadata %>% 
  semi_join(., genome_counts_filt, by = "genome") %>% 
  arrange(match(genome,genome_counts_filt$genome))

genome_tree <- keep.tip(genome_tree, tip=genome_metadata$genome) # keep only MAG tips

#load("data/genome_gifts.Rdata")
```

## Taxonomy overview 

### Stacked barplot

```{r taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(. ~ host_order.x+host_species.x,  scales="free") + #facet 
    guides(fill = guide_legend(ncol = 1)) +
    theme(
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "lightgrey"),
          strip.text = element_text(size = 12, lineheight = 0.6),
          strip.placement = "outside",
          axis.text.x = element_blank(), axis.ticks.x = element_blank(),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

**Number of bacteria phyla**

```{r phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Bacteria phyla in squamates**

```{r phyla_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}

squamates <- sample_metadata %>% 
  filter(sample != "EHI01806" & sample != "EHI01812") %>%
  filter(host_order=="Squamata") %>% 
  dplyr::select(sample) %>% 
  pull()

squamate_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(squamates)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% squamate_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

**Bacterial phyla in rodents**

```{r phyla_cap, comment="", echo=FALSE, message=FALSE, warning=FALSE}

rodents <- sample_metadata %>% 
  filter(host_order=="Rodentia") %>% 
  dplyr::select(sample) %>% 
  pull()

rodent_genomes <- genome_counts %>% 
  column_to_rownames("genome") %>% 
  select(all_of(rodents)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% rodent_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```



**Number of Archaeal phyla**

```{r arch, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Archaea phyla in squamates**

```{r arch_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(genome %in% squamate_genomes) %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()
```

**Archaea phyla in rodents**

```{r arch_pre, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(genome %in% rodent_genomes) %>% 
  filter(domain == "d__Archaea") %>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull()
```


### Genus and species annotation

**Number of MAGs without species-level annotation**
```{r nonspe, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n())%>%
  select(Mag_nospecies) %>% 
  pull()

```
```{r nonspe_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(species == "s__") %>%
  group_by(phylum) %>%
  summarize(count_nospecies = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>% 
  mutate(percentage=100*count_nospecies/count_total) %>% 
  tt()

```

**Percentage of MAGs without species-level annotation**
```{r sp_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nmags <- nrow(genome_counts)
nonspecies <- genome_metadata %>%
    filter(species == "s__") %>%
    nrow()
perct <- nonspecies*100/nmags
perct
```

**Number of MAGs without genera-level annotation**
```{r nongenera, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nongenera <- genome_metadata %>%
    filter(genus == "g__") %>%
    nrow()
cat(nongenera)
```


### Phylum relative abundances

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum,region, host_order.x) %>%
  summarise(relabun=sum(count))
```

```{r taxonomy_boxplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=mean(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    filter(phylum %in% phylum_arrange) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange))) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[rev(phylum_arrange)]) +
        geom_jitter(alpha=0.5) + 
        theme_minimal() + 
        theme(legend.position="none") +
        labs(y="Phylum",x="Relative abundance")
```

#### Origin: squamate vs rodent
```{r taxonomy_phylum_summary_origin, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              squamate_mean=mean(relabun[host_order.x=="Squamate"]*100, na.rm=T),
              squamate_sd=sd(relabun[host_order.x=="Squamate"]*100, na.rm=T),
              rodent_mean=mean(relabun[host_order.x=="Rodentia"]*100, na.rm=T),
              rodent_sd=sd(relabun[host_order.x=="Rodentia"]*100, na.rm=T)) %>%
    mutate(total=str_c(round(total_mean,3),"Â±",round(total_sd,3)),
           squamate=str_c(round(squamate_mean,3),"Â±",round(squamate_sd,3)),
           rodent=str_c(round(rodent_mean,3),"Â±",round(rodent_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,total,squamate,rodent)
```



```{r taxonomy_jitterplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    left_join(genome_metadata %>% select(phylum,phylum) %>% unique(),by=join_by(phylum==phylum)) %>%
#    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    filter(phylum %in% phylum_arrange[1:20]) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~host_order.x)+
        theme_minimal() + 
        labs(y="phylum", x="Relative abundance", color="Phylum")
```

## Taxonomy boxplot

### Family

```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS normalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample, family, host_order.x) %>%
  summarise(relabun=sum(count))
```

#### Family: squamate vs rodent
```{r taxonomy_family_summary_origin, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              squamate_mean=mean(relabun[host_order.x=="Squamata"]*100, na.rm=T),
              squamate_sd=sd(relabun[host_order.x=="Squamata"]*100, na.rm=T),
              rodent_mean=mean(relabun[host_order.x=="Rodentia"]*100, na.rm=T),
              rodent_sd=sd(relabun[host_order.x=="Rodentia"]*100, na.rm=T))  %>%
    mutate(Total=str_c(round(total_mean,2),"Â±",round(total_sd,2)),
           squamate=str_c(round(squamate_mean,2),"Â±",round(squamate_sd,2)),
           rodent=str_c(round(rodent_mean,2),"Â±",round(rodent_sd,2))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(family,Total,squamate,rodent) %>% 
    paged_table()
```


```{r taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~host_order.x)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")
```

### Genus

```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample, phylum, genus, host_order.x) %>%
  summarise(relabun=sum(count)) %>%
  filter(genus != "g__") %>%
  mutate(genus= sub("^g__", "", genus))
```

### origin and diet
```{r taxonomy_genera_summary_oridiet, warning=FALSE, comments="", message=FALSE}
genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              squamate_mean=mean(relabun[host_order.x=="Squamata"]*100, na.rm=T),
              squamate_sd=sd(relabun[host_order.x=="Squamata"]*100, na.rm=T),
              rodent_mean=mean(relabun[host_order.x=="Rodentia"]*100, na.rm=T),
              rodent_sd=sd(relabun[host_order.x=="Rodentia"]*100, na.rm=T)
              ) %>%
    mutate(Total=str_c(round(total_mean,2),"Â±",round(total_sd,2)),
           squamate=str_c(round(squamate_mean,2),"Â±",round(squamate_sd,2)),
           rodent=str_c(round(rodent_mean,2),"Â±",round(rodent_sd,2))
           ) %>%
    arrange(-total_mean) %>% 
    dplyr::select(genus,Total,squamate,rodent) %>% 
    paged_table()
```

```{r taxonomy_jitterplot_genus, fig.height=14, fig.width=10, fig.fullwidth=TRUE}
genus_arrange <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=sum(relabun)) %>%
    filter(genus != "g__")%>%
    arrange(-mean) %>%
    select(genus) %>%
    mutate(genus= sub("^g__", "", genus)) %>%
    pull()

genus_summary_sort <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=T),sd=sd(relabun, na.rm=T)) %>%
    arrange(-mean) 

genus_summary %>%
  mutate(genus=factor(genus, levels=rev(genus_summary_sort %>% pull(genus)))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
  scale_color_manual(values=phylum_colors) +
  geom_jitter(alpha=0.5) + 
  facet_grid(.~host_order.x)+
  theme_minimal() + 
  theme(axis.text.y = element_text(size=6))+
  labs(y="Family", x="Relative abundance", color="Phylum")

```


<!--chapter:end:04_community_composition.Rmd-->

# Alpha diversity

```{r load_data_alpha, comment="", message=FALSE, warning=FALSE}
load("data/data.Rdata")
load("data/alpha.Rdata")
```

## Summary table

```{r alpha_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
# Calculate Hill numbers
richness <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 0) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(richness = 1) %>%
  rownames_to_column(var = "sample")

neutral <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(neutral = 1) %>%
  rownames_to_column(var = "sample")

phylogenetic <- genome_counts_filt %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, tree = genome_tree) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(phylogenetic = 1) %>%
  rownames_to_column(var = "sample")

# # Aggregate basal GIFT into elements
genome_counts_filt <- genome_counts_filt[genome_counts_filt$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts <- genome_gifts[, colSums(genome_gifts != 0) > 0]

dist <- genome_gifts %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

functional <- genome_counts_filt %>%
  filter(genome %in% rownames(dist)) %>%
  column_to_rownames(var = "genome") %>%
  dplyr::select(where(~ !all(. == 0))) %>%
  hilldiv(., q = 1, dist = dist) %>%
  t() %>%
  as.data.frame() %>%
  dplyr::rename(functional = 1) %>%
  rownames_to_column(var = "sample") %>%
  mutate(functional = if_else(is.nan(functional), 1, functional))

# Merge all metrics
alpha_div <- richness %>%
  full_join(neutral, by = join_by(sample == sample)) %>%
  full_join(phylogenetic, by = join_by(sample == sample))%>%
  full_join(functional, by = join_by(sample == sample))
```
```{r wrap_alpha, warning=FALSE, comments="", message=FALSE, eval=FALSE}
save(alpha_div, file = "data/alpha.Rdata")
```


```{r alpha_div_diets_summary_all, comment="",echo=FALSE, message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "alpha", values_to = "value") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
    group_by(alpha)%>%
    summarise(
              squamate_mean=mean(value[host_order=="Squamata"], na.rm=T),
              squamate_sd=sd(value[host_order=="Squamata"], na.rm=T),
              rodent_mean=mean(value[host_order=="Rodentia"], na.rm=T),
              rodent_sd=sd(value[host_order=="Rodentia"], na.rm=T)) %>%
    mutate(
           Squamata=str_c(round(squamate_mean,2),"Â±",round(squamate_sd,2)),
           Rodentia=str_c(round(rodent_mean,2),"Â±",round(rodent_sd,2))) %>% 
    dplyr::select(alpha,Squamata,Rodentia) %>% 
    tt()
```

## Squamata vs Rodentia

### Shapiro test

```{r alpha_div_shapiro, comment="", message=FALSE, warning=FALSE}
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>% 
  summarize(var.test_p_value_phylo = var.test(value ~ host_order)$p.value) 
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>% 
  summarize(shapiro_p_value = shapiro.test(value)$p.value) %>%
  pull(shapiro_p_value)
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>% 
  summarize(var.test_p_functional = var.test(value ~ host_order)$p.value) 
```

### Plots

```{r alpha_div_boxplot, comment="",echo=FALSE, message=FALSE, warning=FALSE}
#Richness
plot1 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="richness") %>%
  ggplot(aes(y = value, x = host_order, group=host_order, color=host_order, fill=host_order)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(method = "wilcox.test", show.legend = F, size = 3, label.y = c(850), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Host order", y = "Richness")

#Neutral
plot2 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="neutral") %>%
  ggplot(aes(y = value, x = host_order, group=host_order, color=host_order, fill=host_order)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(method = "wilcox.test",show.legend = F, size = 3, label.y = c(150), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  coord_cartesian(ylim = c(0, 150)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Host order", y = "Neutral")

#Phylogenetic
plot3 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="phylogenetic") %>%
  ggplot(aes(y = value, x = host_order, group=host_order, color=host_order, fill=host_order)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(11), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Host order", y = "Phylogenetic")

#Functional
plot4 <- alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric=="functional") %>%
  ggplot(aes(y = value, x = host_order, group=host_order, color=host_order, fill=host_order)) +
  geom_jitter(width = 0.2, show.legend = FALSE) +
  geom_boxplot(width = 0.5, alpha=0.5,outlier.shape = NA, show.legend = FALSE) +
  stat_compare_means(method = "t.test", show.legend = F, size = 3, label.y = c(1.6), label.x = c(1.5))+
  coord_cartesian(xlim = c(1, NA)) +
  theme_classic() +
  theme(
    strip.background = element_blank(),
    panel.grid.minor.x = element_line(size = .1, color = "grey"),
    axis.text = element_text(size=10),
    axis.title.y = element_text(margin = margin(t = 0, r = 20, b = 0, l = 0)),
    axis.title.x = element_text(margin = margin(t = 20, r = 0, b = 0, l = 0))
    )+
  labs(x = "Host order", y = "Functional")

```

```{r div_plot_together, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot1,plot2,plot3, plot4, ncol = 2))
```
```{r div_plot_2, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}
grid.arrange(arrangeGrob(plot2,plot3))
```


```{r richness_species, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}

ggplotly(
  
alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric == "richness") %>%
  ggplot(aes(x = host_species, y = value, colour = host_species,
             label = sample, label2 = singlem_fraction, label3 = host_reads)) +
  geom_jitter(size = 2, alpha = 0.5, height = 0, width = 0.1) +
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  scale_y_continuous(limits = c(0, 250)) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(y = "Richness Hill diversity")

) 

```

```{r phylo_species, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}

ggplotly(

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric == "phylogenetic") %>%
  ggplot(aes(x = host_species, y = value, colour = host_species,
             label = sample, label2 = singlem_fraction, label3 = host_reads)) +
  geom_jitter(size = 2, alpha = 0.5, height = 0, width = 0.1) +
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  scale_y_continuous(limits = c(0, 10)) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(y = "Phylogenetic Hill diversity")

) 
```

```{r neutral_species, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}

ggplotly(

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric == "neutral") %>%
  ggplot(aes(x = host_species, y = value, colour = host_species,
             label = sample, label2 = singlem_fraction, label3 = host_reads)) +
  geom_jitter(size = 2, alpha = 0.5, height = 0, width = 0.1) +
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  scale_y_continuous(limits = c(0, 120)) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(y = "Neutral Hill diversity")

) 
```


```{r func_species, comment="", echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=8, fig.fullwidth=TRUE}

ggplotly(

alpha_div %>%
  pivot_longer(-sample, names_to = "metric", values_to = "value") %>%
  left_join(., sample_metadata, by = join_by(sample == sample)) %>%
  filter(metric == "functional") %>%
  ggplot(aes(x = host_species, y = value, colour = host_species,
             label = sample, label2 = singlem_fraction, label3 = host_reads)) +
  geom_jitter(size = 2, alpha = 0.5, height = 0, width = 0.1) +
  stat_summary(fun = "mean", geom = "crossbar", colour = "black", width = 0.4) +
  facet_wrap(~host_order, scales = "free") +
  scale_colour_manual(values = species_colours) +
  scale_y_continuous(limits = c(0, 2)) +
  theme_classic() +
  theme(
    legend.position = "none",
    axis.text.x = element_text(face = "italic", angle = 90, vjust = 0.5, hjust = 1)
  ) +
  labs(y = "Functional Hill diversity")

)
```

<!--chapter:end:05_alpha_diversity.Rmd-->

# Beta diversity

```{r load_data_beta, comment="", message=FALSE, warning=FALSE, eval=TRUE}
load("data/data.Rdata")
load("data/beta.Rdata")
```

#### Calculate beta Hill div

```{r beta_div, comment="", message=FALSE, warning=FALSE, eval=FALSE}
beta_q0n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 0)

beta_q1n <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0)) %>%
  hillpair(., q = 1)

genome_counts_filt <- genome_counts_filt %>%
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_tree <- keep.tip(genome_tree, tip=genome_counts_filt$genome)

# beta_q1p <- genome_counts_filt %>%
#   column_to_rownames(., "genome") %>%
#   filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
#   select_if(~!all(. == 0)) %>%
#   hillpair(., q = 1, tree = genome_tree)

genome_counts_filt <- genome_counts[genome_counts$genome %in% rownames(genome_gifts),]
genome_counts_filt <- genome_counts_filt %>%
  remove_rownames() %>% 
  column_to_rownames(., "genome") %>%
  filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
  select_if(~!all(. == 0))%>%
  rownames_to_column(., "genome")

genome_gifts1 <- genome_gifts[rownames(genome_gifts) %in% genome_counts_filt$genome,]
genome_gifts1 <- genome_gifts1[, colSums(genome_gifts1 != 0) > 0]

dist <- genome_gifts1 %>%
  to.elements(., GIFT_db) %>%
  traits2dist(., method = "gower")

# beta_q1f <- genome_counts_filt %>%
# #  remove_rownames() %>% 
#   column_to_rownames(., "genome") %>%
#   filter(rowSums(. != 0, na.rm = TRUE) > 0) %>%
#   select_if(~!all(. == 0)) %>%
#   hillpair(., q = 1, dist = dist)

```
#### Save outputs
```{r save_beta, comment="", message=FALSE,echo=FALSE,warning=FALSE, eval=FALSE}
save(beta_q0n, 
     beta_q1n, 
     # beta_q1p, 
     # beta_q1f,
     genome_counts,
     sample_metadata,
     genome_metadata,
     genome_gifts,
     genome_gifts1,
     genome_tree,
     file = "data/beta.Rdata")
```

### Permanova analysis
```{r}
set.seed(1234)
```

#### Richness
```{r permanova_q0, comment="", message=FALSE, warning=FALSE}
sample_metadata_row <- column_to_rownames(sample_metadata, "sample") 
sample_metadata_row <- sample_metadata_row[labels(beta_q0n$S), ]
sample_metadata_trim <- sample_metadata %>%
  filter(sample != "EHI01806" & sample != "EHI01812")

betadisper(beta_q0n$S, sample_metadata_row$host_order) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q0n$S ~ host_order, 
        data = sample_metadata_trim %>% arrange(match(sample,labels(beta_q0n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

#### Neutral
```{r permanova_wildcap_q1, comment="", message=FALSE, warning=FALSE}
betadisper(beta_q1n$S, sample_metadata_row$host_order) %>% permutest(., pairwise = TRUE) 

adonis2(beta_q1n$S ~ host_order, 
        data = sample_metadata_trim %>% arrange(match(sample,labels(beta_q1n$S))), 
        permutations = 999) %>%
        broom::tidy() %>%
        tt()
```

#### Phylogenetic
```{r permanova_wildcap_qP, comment="", message=FALSE, warning=FALSE}
# betadisper(beta_q1p$S, sample_metadata_row$host_order) %>% permutest(., pairwise = TRUE) 
# adonis2(beta_q1p$S ~ host_order, 
#         data = sample_metadata %>% arrange(match(sample,labels(beta_q1p$S))), 
#         permutations = 999) %>%
#         broom::tidy() %>%
#         tt()
```

#### Functional
```{r permanova_wildcap_qF, comment="", message=FALSE, warning=FALSE}
# betadisper(beta_q1f$S, sample_metadata_row$host_order) %>% permutest(., pairwise = TRUE) 
# adonis2(beta_q1f$S ~ host_order,
#         data = sample_metadata %>% arrange(match(sample,labels(beta_q1f$S))),
#         permutations = 999) %>%
#         broom::tidy() %>%
#         tt()
```

### Beta diversity plots

#### Richness

```{r beta_div_nmds_neutral_plot_median, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}

ggplotly(

beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_order) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_order,
             label = sample, label1 = singlem_fraction)) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) 

) 
```
#### Richness (host species)

```{r beta_div_nmds_richness_plot_median_species, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}

ggplotly(

beta_q0n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species, shape = host_order,
             label = sample, label1 = singlem_fraction)) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    )

) 
```

#### Neutral

```{r beta_div_neutral, comment="", echo=FALSE, message=FALSE, warning=FALSE}

ggplotly(

beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_order) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_order,
             label = sample, label1 = singlem_fraction)) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    ) 
# +geom_text_repel(aes(label = individual), size=3)

) 
```
#### Neutral (host species)

```{r beta_div_nmds_neutral_plot_median_species, comment="", message=FALSE, warning=FALSE, fig.height=7, fig.width=10, fig.fullwidth=TRUE, eval=FALSE}

ggplotly(

beta_q1n$S %>%
  vegan::metaMDS(., trymax = 500, k = 2, trace=0) %>%
  vegan::scores() %>%
  as_tibble(., rownames = "sample") %>%
  dplyr::left_join(sample_metadata, by = join_by(sample == sample)) %>%
  group_by(host_species) %>%
  mutate(x_cen = mean(NMDS1, na.rm = TRUE)) %>%
  mutate(y_cen = mean(NMDS2, na.rm = TRUE)) %>%
  ungroup() %>%
  ggplot(aes(x = NMDS1, y = NMDS2, color = host_species, shape = host_order,
             label = sample, label1 = singlem_fraction)) +
    geom_point(size = 4) +
    #   stat_ellipse(aes(color = beta_q1n_nmds$Groups))+
    geom_segment(aes(x = x_cen, y = y_cen, xend = NMDS1, yend = NMDS2), alpha = 0.9, show.legend = FALSE) +
    theme_classic() +
    theme(
      axis.text.x = element_text(size = 12),
      axis.text.y = element_text(size = 12),
      axis.title = element_text(size = 12, face = "bold"),
      axis.text = element_text(face = "bold", size = 12),
      panel.background = element_blank(),
      axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
      legend.text = element_text(size = 12),
      legend.title = element_text(size = 14),
      legend.position = "right", legend.box = "vertical"
    )

)

```




<!--chapter:end:06_beta_diversity.Rmd-->

